src/Designer/backend/src/Designer/Middleware/UserRequestSynchronization/RepoUserWide/DistributedLockProviderExtensions.cs

```diff
@@ -0,0 +1,37 @@
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Altinn.Studio.Designer.Models;
+using Medallion.Threading;
+
+namespace Altinn.Studio.Designer.Middleware.UserRequestSynchronization.RepoUserWide;
+
+/// <summary>
+/// Extension methods for <see cref="IDistributedLockProvider"/>.
+/// Enriches the provider with methods that create locks based on <see cref="AltinnRepoEditingContext"/>.
+/// </summary>
+public static class DistributedLockProviderExtensions
```

Comment:
Don't remember specifically what, but there was a bug in this library (DistributedLock library) that I couldn't work out when I worked on AppsMonitoring. I ended up with a custom implementation in this file: https://github.com/Altinn/altinn-tools/blob/a36cdaa9c9790549973bf871e98baf5b139322ae/AppsMonitoring/src/Altinn.Apps.Monitoring/Application/Db/DistributedLocking.cs

Just commenting for knowledge sharing, not a request for change